pages/gallery.tsx

```diff
@@ -1,14 +1,91 @@
-import Link from 'next/link'
 import Layout from '@/components/Layout/Layout'
+import styles from 'styles/gallery.module.scss'
+import { useState, useEffect } from 'react'
 
 export default function Gallery() {
+	const [shownPopup, setShownPopup] = useState(false)
+	const [popupImg, setPopupImg] = useState('')
+	const [currentPage, setCurrentPage] = useState(1)
+	const [itemsPerPage, setItemsPerPage] = useState(24)
+
+	const images = Array.from({ length: 24 }, (_, i) => `images/gallery/${i + 1}.jpg`)
+	const indexOfLastItem = currentPage * itemsPerPage
+	const indexOfFirstItem = indexOfLastItem - itemsPerPage
+	const currentImages = images.slice(indexOfFirstItem, indexOfLastItem)
+
+	useEffect(() => {
+		const handleResize = () => {
+			if (window.innerWidth >= 1867) {
+				setItemsPerPage(24)
+			} else if (window.innerWidth >= 1632) {
+				setItemsPerPage(21)
+			} else if (window.innerWidth >= 1400) {
+				setItemsPerPage(18)
+			} else if (window.innerWidth >= 1166) {
+				setItemsPerPage(15)
+			} else if (window.innerWidth >= 934) {
+				setItemsPerPage(12)
+			} else if (window.innerWidth >= 700) {
+				setItemsPerPage(9)
+			} else if (window.innerWidth >= 466) {
+				setItemsPerPage(6)
+			} else {
+				setItemsPerPage(3)
+			}
+		}
+
+		handleResize()
+
+		window.addEventListener('resize', handleResize)
+
+		return () => {
+			window.removeEventListener('resize', handleResize)
+		}
+	}, [])
+
+	function handleTogglePopup(event) {
+		const { src } = event.target
+		setShownPopup(true)
+		setPopupImg(src)
+	}
```

Comment:
this could be improved with ts like:
```suggestion
	function handleTogglePopup(e: React.MouseEvent<HTMLImageElement>) {
          const target = e.target as HTMLImageElement;
          if (target && target.src) {
            setShownPopup(true);
            setPopupImg(target.src);
          }
        }
```